pycbc/psd/analytical.py

```diff
@@ -85,15 +61,24 @@ def from_string(psd_name, length, delta_f, low_freq_cutoff):
         Frequency resolution of the frequency series.
     low_freq_cutoff : float
         Frequencies below this value are set to zero.
-    
+
     Returns
     -------
     psd : FrequencySeries
         The generated frequency series.
     """
+    if psd_name not in _psd_list:
+        raise ValueError(psd_name + ' not found among LALSimulation PSD functions.')
+    kmin = int(low_freq_cutoff / delta_f)
+    lalseries = lal.CreateREAL8FrequencySeries(
+        '', lal.LIGOTimeGPS(0), 0, delta_f, lal.DimensionlessUnit, length)
     try:
-        func = lalsimulation.__dict__[_name_prefix + psd_name]
+        func = lalsimulation.__dict__[_name_prefix + psd_name + _name_suffix]
     except KeyError:
-        raise ValueError(psd_name + ' not found among LALSimulation PSD functions.')
-    return from_lalsimulation(func, length, delta_f, low_freq_cutoff)
-
+        func = lalsimulation.__dict__[_name_prefix + psd_name]
+        func(lalseries, low_freq_cutoff)
```

Comment:
In fact I think this will not work, because `func` would take just a float frequency in this case, not a series plus a cutoff.